### DISCUSSION TITLE
Flowise chatbot (deployed on render cloud) ignoring Vector database

### DISCUSSION THREAD

**Discussion by nag-24** [FlowiseAI/Flowise#1848] [Q&A]
Hi Flowise community, 
I need help with an issue. I’m running my AI chatbot on my website off the github.com/FlowiseAI/Flowise forked from the latest commit (deployed on Render cloud) and embedding the chatbot with the HTML embed code given after creating your chatflow on Flowise. The problem I’m having is that after the recent commits, the chatbot is not referring my pinecone vector database when providing responses to user prompts. It is coming up with it’s own responses with no relation to the vector database. I rechecked the chatflow and everything seems configured. Screenshot given.
I tried a standalone chatflow hosted from localhost and the retrieval is a little better but not comprehensive. 
Please advise what the issue could be – is it a problem with render cloud. I checked the status and it seemed fine. 
![Screenshot of Chatflow](https://github.com/FlowiseAI/Flowise/assets/160205830/117ca730-024b-428b-b99b-e2f18507896f)

**Comment by jbeltran73-2** [comment]
I'm confused. Your vector database doesn't have any connection with any documents. Why do you use the vector database then for?
Also, for using embeddings, I recommend the 3-large. Although is more expensive it makes the difference.

**Reply by SmartMart69** [reply]
pinecone does not need a document once the vector database has been created (then with a document of course).

**Reply by jbeltran73-2** [reply]
Yes, that could be, of course.

**Comment by SmartMart69** [comment]
Maybe starting from scratch (including building a new vector database in pinecone) would do the trick?

**Comment by nag-24** [comment]
Hi, thanks for your inputs. I think I figured what the problem is. The Flowise configuration is sending the user message + chat history as a standalone question/query to the retrieval database and then sending the retrieval docs + user message + chat history +general chatbot instructions to OpenAI's API to get the response. The problem here is that the general chatbot instructions also need to be incorporated in the query for retrieval otherwise the retrieval is ineffective. My hunch is that this will affect a lot of other use cases as well. I will need to do some custom coding I guess to fix this configuration, but if other users also face this problem, it would be worth considering this as a feature request. Pls share your views/experiences. I will post this on discord as well.

**Reply by HenryHengZJ** [reply]
you can combine the intrsuction + user message + chat history, to rephrase the question:
<img width="824" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/26460777/052ce451-f14f-4b04-94bd-43df3c61c81f">

**Reply by nag-24** [reply]
![Screenshot of Chatflow_05-Mar-24](https://github.com/FlowiseAI/Flowise/assets/160205830/adb45423-63b7-4ad7-834d-d2dd6b4f7c84)
![Screenshot Parameters for Conversational Retrieval QA Chain](https://github.com/FlowiseAI/Flowise/assets/160205830/f95c79b5-2943-46fd-89b5-dc1a291d124c)
Thanks Henry. I tried this but I'm getting the result as "Missing value for input context". Pls see above the screenshots. I also tried to modify the code for github/com/FlowiseAI/Flowise/packages/components/nodes/chains/ConversationalRetrievalQAChain/ConversationalRetrievalQAChain.ts
in my forked repository - you can see it here https://github.com/FlowiseAI/Flowise/compare/main...MyAIAdvisor:Flowise_MyAIAdv:main
Pls let us know what you think is the best way to handle this.
